runtime/astar/src/lib.rs

```diff
@@ -1933,22 +1933,23 @@ impl_runtime_apis! {
         }
 
         fn query_weight_to_asset_fee(weight: Weight, asset: VersionedAssetId) -> Result<u128, XcmPaymentApiError> {
-            match asset.try_as::<XcmAssetId>() {
-                // for native token
-                Ok(asset_id) if asset_id.0 == xcm_config::AstarLocation::get() => {
-                    Ok(WeightToFee::weight_to_fee(&weight))
-                }
-                // for foreign assets that have 'units per second' configured
-                Ok(asset_id) => {
-                    let versioned_location = VersionedLocation::V4(asset_id.0.clone());
+            let asset = asset.into_version(xcm::v4::VERSION).map_err(|_| XcmPaymentApiError::VersionedConversionFailed)?;
+            let asset_id: XcmAssetId = asset.try_into().map_err(|_| XcmPaymentApiError::VersionedConversionFailed)?;
+
+            // for native token
+            if asset_id.0 == xcm_config::AstarLocation::get() {
+                Ok(WeightToFee::weight_to_fee(&weight))
+            }
+            // for foreign assets that have 'units per second' configures
```

Comment:
I wouldn't usually comment this, but on the left hand side it was _configured_, but now it's _configures_ 🙃